addons/udes_stock_refactoring/models/stock_move.py

```diff
@@ -0,0 +1,437 @@
+           equivalent moves after they have been transformed.
+           The output moves may be identical to the input, may contain none
+           of the input moves, or anywhere in between.
+           The output should contain a functionally similar set of moves.
+        """
+        if stage is not None and stage not in U_STOCK_REFACTOR_STAGES.values():
+            raise UserError(_("Unknown stage for move refactor: %s") % stage)
+        moves = self
+
+        if self._context.get("disable_move_refactor"):
+            return moves
+
+        rf_moves = moves.filtered(
+            lambda m: m.picking_type_id and m.state not in ["draft", "cancel"]
+        )
+
+        if stage is not None:
+            rf_moves = rf_moves.filtered(lambda m: U_STOCK_REFACTOR_STAGES[m.state] == stage)
+
+        for picking_type, pt_moves in rf_moves.groupby("picking_type_id"):
+            for stage, st_moves in pt_moves.groupby(lambda m: U_STOCK_REFACTOR_STAGES[m.state]):
+                if stage == "confirm":
+                    action = picking_type.u_post_confirm_action
+                elif stage == "assign":
+                    action = picking_type.u_post_assign_action
+                elif stage == "validate":
+                    action = picking_type.u_post_validate_action
+                else:
+                    continue  # Don't refactor cancel or draft moves.
+
+                if action:
+                    _logger.info(
+                        "Refactoring %s at %s using %s: %s",
+                        picking_type.name,
+                        stage,
+                        action,
+                        st_moves.ids,
+                    )
+                    func = getattr(st_moves, "refactor_action_" + action)
+                    new_moves = func()
+                    if new_moves is not None:
+                        moves -= st_moves
+                        moves |= new_moves
+
+        return moves
+
+    def _action_confirm(self, *args, **kwargs):
+        """Extend _action_confirm to trigger refactor action.
+
+        Odoos move._action_confirm returns all the moves passed in to it, after
+        merging any it can. In places the return value is used to immediately
+        assign, so any created moves should be returned.
+
+        However, in stock.move._split() in Odoo core, _action_confirm is
+        called on a single move and it expects a single move to be returned
+        (will error out if not). Therefore the refactor at confirm should avoid
+        splitting/creating moves. Luckily this is unlikely to be an issue as
+        it makes little sense to split moves at this stage, before they have
+        stock reserved against them.
+        """
+        res = super(StockMove, self)._action_confirm(*args, **kwargs)
+        post_refactor_moves = res._action_refactor(stage="confirm")
+
+        if post_refactor_moves != res:
+            raise UserError(
```

Comment:
Is this error still valid?

Reply:
I believe it is as the `mrp` module is still part of core Odoo in 13.